src/pool.ts

```diff
@@ -278,7 +279,7 @@ export class PoolConnectionProxy {
     connection._setProxy(this);
   }
 
-  public get connection(): AwaitConnection {
+  get connection(): AwaitConnection {
```

Comment:
This is still public (IIRC `public` is implicit, and to make `private` you always add `private` explicitly)

Reply:
It shouldn't be possible for users to unwrap real connections here, so I'd say this should be private.

Reply:
I handled this a bit differently than how it is done in Python. In the Python driver, the metaclass `PoolConnectionProxyMeta` is used to copy all methods from the connection class into the proxy class. `acquire` returns a proxy, `release` accepts a proxy (this is the same in both drivers). So if the user acquires a proxy, it will behave like a connection object while not exposing the underlying connection.

I know this kind of dynamism is possible in JavaScript and TypeScript, but I wasn't sure whether this is a common approach when working with TS. I kept the proxy class simpler, and exposed the connection.

The difference in practice is:
```python
    async def execute(self, query):
        async with self.acquire() as con: # <-- here "con" is a connection proxy
            return await con.execute(query)
```

```ts
  async run<T>(
    action: (connection: AwaitConnection) => Promise<T>
  ): Promise<T> {
    const proxy = await this.acquire();

    try {
      return await action(proxy.connection);  // <-- here the actual connection is used to execute
    } finally {
      await this.release(proxy);
    }
  }

  async execute(query: string): Promise<void> {
    return await this.run(async (connection) => {
      return await connection.execute(query);
    });
  }
```

I wanted to speak with you about this detail, then I forgot while asking about other details.
If you prefer to make the connection private, then I propose to add the 5 methods to the proxy class (fetchOne, fetchOneJSON, fetchAll, fetchAllJSON, execute).

Reply:
> I kept the proxy class simpler, and exposed its connection.

Yeah, we have to follow Python's design here.

> In the Python driver, the metaclass PoolConnectionProxyMeta is used to copy all methods from the connection class into the proxy class.  [...]  I know this kind of dynamism is possible in JavaScript and TypeScript, but I wasn't sure whether this is a common approach when working with TS. 

We don't need to replicate metaclasses or employ other magic, like introspecting prototypes. The methods can simply be hardcoded. The `ProxyConnection` type should just implement `IConnection` interface, which will give us a heads up when we add new APIs and forget to add them to `ProxyConnection`.

The connection should be private. The whole point is that it's impossible for the user to accidentally use a released pool connection. If the connection is exposed, nothing precludes the user from storing a reference to the wrapped connection object and using it later when it's acquired by some other call site.

Reply:
I modified the code to keep the connection private. :+1: I also made the queue private, and finally included a `PoolStats` returning the list of pending consumers + number of open connections. We will probably do a second pass over the pool stats when we have occasion to focus on this subject.